feat(integrations/gmail): fix gmail oauth flow + retry slackbox

quu-ack opened the pull request:
This PR reworks the OAuth and manual creation flows.
Previously, when a non-expired token was changed (for example, to switch the user account for Gmail access), the flow would always reuse the token stored in the state instead of the newly provided one.
This PR fixes that behavior by ensuring the updated token is correctly used.

Concerning Slackbox, we ensure callAction resilience by adding a retry logic

quu-ack committed:
feat(bots/slackbox): create bot

quu-ack committed:
Merge branch 'master' into quack/create-slackbox

quu-ack committed:
fix pipeline

quu-ack committed:
Merge branch 'quack/create-slackbox' of https://github.com/botpress/botpress into quack/create-slackbox

quu-ack committed:
chore: update pnpm-lock.yaml for slackbox dependencies

quu-ack committed:
fix pipeline

quu-ack committed:
Merge branch 'master' into quack/create-slackbox

quu-ack committed:
Merge branch 'master' into quack/create-slackbox

quu-ack committed:
fix pr comments

quu-ack committed:
wip test gmail

quu-ack committed:
refactor(integrations/gmail): clean up debug code and simplify OAuth/conversation handling

quu-ack committed:
Merge remote-tracking branch 'origin/master' into quack/create-slackbox

quu-ack committed:
Merge branch 'master' into quack/create-slackbox

quu-ack committed:
fix ci

quu-ack committed:
Merge branch 'quack/create-slackbox' of https://github.com/botpress/botpress into quack/create-slackbox

quu-ack committed:
Merge branch 'master' into quack/create-slackbox

quu-ack committed:
add logs

Math-Fauch commented on integrations/gmail/src/setup.ts:
```diff
@@ -1,32 +1,69 @@
+  if (ctx.configurationType !== 'customApp') {
+    logger.forBot().info('Using refresh token from configuration')
+    googleClient = await createFromRefreshToken()
+  } else if (ctx.configuration.oauthAuthorizationCode) {
```
**issue:** this `if else` should only be on `ctx.configurationType`. You can maybe add a check for empty string for `ctx.configuration.oauthAuthorizationCode` inside the else statement.

Math-Fauch reviewed (changes requested):
The CI does not pass.
Can you clarify the reason why we need to fix the Gmail OAuth flow?

quu-ack commented on integrations/gmail/src/google-api/google-client.ts:
```diff
@@ -83,30 +95,60 @@ export class GoogleClient {
-      id: ctx.integrationId,
-      payload: { refreshToken },
-    })
+    // Use patchState to preserve existing fields like lastHistoryId
```
I just wanted to explain why we are using a patchstate instead of a setstate here, and then comment why we are using a getorsetstate after - not sure it would be clear for everyone

quu-ack committed:
rework auth branching

quu-ack committed:
fix slackbox

quu-ack committed:
Fix slackbox retry logic for Slack channel conversation

Add retry mechanism (3 attempts with 2s delay) when fetching Slack conversation ID to handle timeout issues caused by large workspace
channel pagination

quu-ack commented on bots/slackbox/src/index.ts:
```diff
@@ -17,36 +17,14 @@ bot.on.message('*', async (props) => {
+    const response = await client.callAction({
+      type: 'slack:startChannelConversation',
+      input: {
+        channelName: DEFAULT_SLACK_CHANNEL,
```
getOrSetState doesn't work, I replaced this with a retry and it works perfectly

Math-Fauch commented on bots/slackbox/src/index.ts:
```diff
@@ -5,49 +5,51 @@ import * as bp from '.botpress'
-
-bot.on.message('*', async (props) => {
-  const { conversation, message, client, ctx, logger } = props
+let cachedSlackConversationId: string | undefined
```
**note:** Since this runs on a serverless environment, this var will probably be dropped between bot calls.
Also I think that changes for the bot does not belong to this PR scope.

quu-ack committed:
remove unnecessary logic

quu-ack committed:
Merge branch 'master' into quack/create-slackbox
